### PR TITLE
[Improvement](runtime-filter) adjust always true judge rule of runtime filter apply

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
adjust always true judge of runtime filter apply
tpch sf1000:
图片

tpcds sf1000:
图片

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by englefly** [comment]
set enable_runtime_filter_prune=false;

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/exprs/vruntimefilter_wrapper.h [hunk 1]
```diff
@@ -39,12 +39,25 @@ class VExprContext;
 } // namespace vectorized
 } // namespace doris
 
+inline double get_in_list_ignore_thredhold(size_t list_size) {
+    return std::log2(list_size + 1) / 64;
+}
+
+inline double get_comparison_ignore_thredhold() {
```
Comment (HappenLee) [inline be/src/vec/exprs/vruntimefilter_wrapper.h:46]: noneed inline should be a constexpr func

File: be/src/vec/exprs/vruntimefilter_wrapper.cpp [hunk 2]
```diff
@@ -81,26 +76,16 @@ void VRuntimeFilterWrapper::close(VExprContext* context,
 
 Status VRuntimeFilterWrapper::execute(VExprContext* context, Block* block, int* result_column_id) {
     DCHECK(_open_finished || _getting_const_col);
+    _judge_counter--;
     if (_always_true) {
         block->insert({create_always_true_column(block->rows(), _data_type->is_nullable()),
```
Comment (HappenLee) [inline be/src/vec/exprs/vruntimefilter_wrapper.cpp:81]: add TODO，should be return const column？

File: be/src/vec/exprs/vruntimefilter_wrapper.h [hunk 3]
```diff
@@ -39,12 +39,25 @@ class VExprContext;
 } // namespace vectorized
 } // namespace doris
 
+inline double get_in_list_ignore_thredhold(size_t list_size) {
```
Comment (HappenLee) [inline be/src/vec/exprs/vruntimefilter_wrapper.h:42]: better eazy to cal and understand

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 4]
```diff
@@ -2176,11 +2176,17 @@ uint16_t SegmentIterator::_evaluate_vectorization_predicate(uint16_t* sel_rowid_
...
     bool all_pred_always_true = true;
     for (const auto& pred : _pre_eval_block_predicate) {
-        if (!pred->always_true()) {
+        if (!pred->always_true(false)) {
             all_pred_always_true = false;
             break;
         }
     }
+    if (all_pred_always_true) {
+        for (const auto& pred : _pre_eval_block_predicate) {
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2185]: why here need set always true ？

File: be/src/vec/exprs/vruntimefilter_wrapper.cpp [hunk 5]
```diff
@@ -81,26 +76,16 @@ void VRuntimeFilterWrapper::close(VExprContext* context,
 
 Status VRuntimeFilterWrapper::execute(VExprContext* context, Block* block, int* result_column_id) {
     DCHECK(_open_finished || _getting_const_col);
+    _judge_counter--;
```
Comment (HappenLee) [inline be/src/vec/exprs/vruntimefilter_wrapper.cpp:79]: the operation of `judge_counter`  is error, must do change

**Review by HappenLee — CHANGES_REQUESTED** [review]
some fatal logic error

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 4 above)
Comment (BiteTheDDDDt) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:2185]: Whenever we process a block of data, we should change the judge counter

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**